### ISSUE TITLE
Possible false positive with nested functions

### ISSUE THREAD

**Issue by nilshg** [tonyhffong/Lint.jl#96]
I often see Lint complaining when I define a function within a function. A minimal example would be:
````
function func1(a)
  x = a^2
  function func2(x)
    x^2
  end
  func2(x)
end
````
This gives an info that `x` was declared but not used, even though it's clearly used in `func2`. Am I getting something wrong here or is this a false positive?

**Comment by tonyhffong** [comment]
I think it may be a false positive. However, don't you think there might be an ambiguity (when a human reads your code) when the inner x could reference the argument x or the variable x within func1, even though it may still run as you (the author) may expect?

**Comment by nilshg** [comment]
You're probably right, and in fact I've solved this warning in some instances in my code by renaming the arguments to `func2`; this is just a little inconvenienct when constructing functions that are explicitly meant to be constructed with the variables created within `func1`. I'm thinking of something like `func2(y, x=x)`, but maybe there just isn't anything Lint can do about this.
